nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server_impl.hpp

```diff
@@ -226,6 +228,9 @@ void BtActionServer<ActionT>::executeCallback()
   // Give server an opportunity to populate the result message or simple give
   // an indication that the action is complete.
   auto result = std::make_shared<typename ActionT::Result>();
+
+  populateErrorCode(result);
```

Comment:
Wouldn't this only be required if the goal failed (e.g. add to L244)?

Reply:
The result goes into this `on_completion_callback_(result, rc);`. I assumed that we should populate the error before this function call.

Reply:
Good idea